### PR TITLE
Improve instructions, adding examples and notes on middleware ordering

### PR DESCRIPTION
**Pull request by qnm**
After discussion in gaearon/redux-thunk#9 I've prepared slightly more elaborate installation instructions based on my current understanding of installation.

Please let me know if I've got anything wrong!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,47 @@ A live-editing time travel environment for Redux.
 npm install --save-dev redux-devtools
 ```
 
+DevTools is a middleware, which should be added to your middleware stack *after* `applyMiddleware` because `applyMiddleware` is potentially asynchronous. Otherwise, DevTools won't see the raw actions emitted by the Promise middleware etc.
```
Comment (gaearon) [inline README.md]: >DevTools is a middleware

It's a “store enhancer”. The result of `applyMiddleware` is also a “store enhancer”.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,47 @@ A live-editing time travel environment for Redux.
...
+Lastly, include the `DebugPanel` in your page:
+```
+export default class App extends Component {
+  render() {
+    return (
+      <Provider store={store}>
+        {() => <CounterApp />}
+      </Provider>
+      <div>
+        <Provider store={store}>
```
Comment (gaearon) [inline README.md]: This looks like extra copy-paste, we don't need `<CounterApp>` twice.

**Comment by qnm** [comment]
@gaearon Sure. Updated.

**Comment by gaearon** [comment]
Sorry, I confused you all over. :-P

“Middleware” is the thing that wraps `dispatch`. [redux-thunk](https://github.com/gaearon/redux-thunk), [redux-promise](https://github.com/acdlite/redux-promise) are all middleware.

“Store enhancer” is the thing that wraps `createStore`. Redux DevTools and `applyMiddleware` are both store enhancers.

Hope it makes more sense.

**Comment by qnm** [comment]
@gaearon Doesn't this need further changes, or will you make them yourself?  I was just about to pick this up again!

**Comment by gaearon** [comment]
I just tweaked it a bit. Thanks!

**Pull request merged**